jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -65,7 +66,7 @@ public void assignTask(Task task, TaskContext context, String excludedUser) {
             return;
         }
  
-        Assignment assignTo = this.strategy.apply(task, context, excludedUser);
+        Assignment assignTo = computeAssignementStrategyForTask(task).apply(task, context, excludedUser);
```

Comment:
See my comment later: as strategy attribute is the global strategy, and here can be potentially modified per task, then the warning log written 2 lines below is not reflecting the current strategy but the global one. Perhaps you can split this line into 2, and store the current strategy in a local variable for being used in that log.

Reply:
fixed